src/Surfnet/ServiceProviderDashboard/Application/ViewObject/EntityOidcConfirmation.php

```diff
@@ -66,7 +66,9 @@ public static function fromEntity(DomainEntity $entity)
      */
     public function getEntityId()
     {
-        if ($this->protocol === DomainEntity::TYPE_OPENID_CONNECT_TNG) {
+        $isOidcng = $this->protocol === DomainEntity::TYPE_OPENID_CONNECT_TNG ||
```

Comment:
Please throw an exception when an entity type isn't supported. That way an exception will, be thrown when an id is requested for an unsupported type. So if this is getting extended this will result in an exception during development which could be acted upon.

Reply:
Making the constructor type safe is a good addition. As that is the only moment you could create this view object for the wrong entity type.

Reply:
Nice!